src/linkify-element.js

```diff
@@ -32,8 +32,11 @@ function tokensToNodes(tokens, opts, doc) {
 
 	for (let i = 0; i < tokens.length; i++) {
 		let token = tokens[i];
+		let validated = options.resolve(opts.validate,
+			token.hasProtocol ? token.hasProtocol() : false,
```

Comment:
I see what you mean. I just find it a pity that information obtained by the tokenizer (e.g. if there is a protocol -and which one-, the domain, the "path") is lost and unaccessible by the functions. It would be nice, in the future, to be able to access some (selected?) information about the token.